README.md

```diff
@@ -452,11 +451,13 @@ decodeJsonAuthor maybeUsername json = do
   obj <- decodeJson json
   author <- obj .: "author"
   following <- obj .: "following"
-  pure $ case maybeUsername of
+  case maybeUsername of
     -- user is logged in and is the author
-    Just (Username username) | author == username -> You
-    -- user is not the author, or no one is logged in, so use the `following` flag
-    otherwise -> author # if following then Following else NotFollowing
+    Just (Username username)
+      | author == username -> Right You
+      -- user is not the author, or no one is logged in, so use the `following` flag
+      | otherwise -> Right $ Other author following
+    Nothing -> Left $ Named "Username" MissingValue
```

Comment:
I don't know if `Named` is the right error to use here. It certainly doesn't seem like `AtKey` is correct either.

Reply:
This isn't quite right; what this example is (supposed to be) getting at is that the `Maybe Username` argument represents the logged-in user, if there is one.

If it's a `Just` then someone is logged in, so we check to see if the logged-in person is the author; if they are the same, then we can return `You`. If they are not the same, then we can return `Other` and check the `following` flag.

If it's a `Nothing` then no one is logged in. In that case the viewer of the article isn't following the author of the article (because they can't be). So `following` should be false, and we can set the return value to `Other author false`.

Maybe this could be cleaned up to better represent the idea that you can have your decoding function depend on an argument without getting too bogged down in the details.

Reply:
I think the point of confusion for me (beyond the initial package mismatch build errors) is that `otherwise` was used outside of the guard.
I'll make another PR to restore this as close as possible to the original example.